File: Restaurant/app/src/main/java/com/example/restaurant/data/model/TakeawayOrder.kt
package com.example.restaurant.data.model

import com.google.gson.annotations.SerializedName
import java.util.Date

/**
 * Model for Takeaway Orders
 */
data class TakeawayOrder(
    @SerializedName("OrderID")
    val id: Int,
    
    @SerializedName("OrderCode")
    val orderCode: String,
    
    @SerializedName("UserID")
    val userId: Int,
    
    @SerializedName("CustomerName")
    val customerName: String,
    
    @SerializedName("Phone")
    val phone: String,
    
    @SerializedName("Address")
    val address: String,
    
    @SerializedName("TotalAmount")
    val totalAmount: Double,
    
    @SerializedName("Status")
    val status: String,
    
    @SerializedName("PaymentMethod")
    val paymentMethod: String,
    
    @SerializedName("PaymentStatus")
    val paymentStatus: String,
    
    @SerializedName("Note")
    val note: String?,
    
    @SerializedName("DeliveryTime")
    val deliveryTime: Date?,
    
    @SerializedName("EstimatedDeliveryTime")
    val estimatedDeliveryTime: Date?,
    
    @SerializedName("CreatedAt")
    val createdAt: String,
    
    @SerializedName("UpdatedAt")
    val updatedAt: String,
    
    @SerializedName("items")
    val items: List<TakeawayOrderItem>? = null
)

/**
 * Model for Takeaway Order Items
 */
data class TakeawayOrderItem(
    @SerializedName("OrderItemID")
    val id: Int,
    
    @SerializedName("OrderID")
    val orderId: Int,
    
    @SerializedName("ItemID")
    val itemId: Int,
    
    @SerializedName("Quantity")
    val quantity: Int,
    
    @SerializedName("Price")
    val price: Double,
    
    @SerializedName("item")
    val menuItem: MenuItem? = null
)

/**
 * Request model to create a new takeaway order
 */
data class TakeawayOrderRequest(
    val customer_name: String,
    val phone: String,
    val address: String,
    val payment_method: String = "Cash on Delivery",
    val note: String? = null,
    val items: List<TakeawayOrderItemRequest>
)

/**
 * Request model for items in a takeaway order
 */
data class TakeawayOrderItemRequest(
    val id: Int,
    val quantity: Int
)

/**
 * Status values for Takeaway Orders
 */
object TakeawayOrderStatus {
    const val PENDING = "Pending"
    const val CONFIRMED = "Confirmed"
    const val IN_PREPARATION = "In Preparation"
    const val OUT_FOR_DELIVERY = "Out for Delivery"
    const val DELIVERED = "Delivered"
    const val CANCELLED = "Cancelled"
}

/**
 * Payment Method values for Takeaway Orders
 */
object TakeawayPaymentMethod {
    const val CASH_ON_DELIVERY = "Cash on Delivery"
    const val ONLINE_PAYMENT = "Online Payment"
}

/**
 * Payment Status values for Takeaway Orders
 */
object TakeawayPaymentStatus {
    const val PENDING = "Pending"
    const val PAID = "Paid"
    const val REFUNDED = "Refunded"
} 

File: hari-restaurant/app/Models/TakeawayOrder.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class TakeawayOrder extends Model
{
    use HasFactory;

    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'takeaway_orders';

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'order_code',
        'user_id',
        'customer_name',
        'phone',
        'address',
        'total_amount',
        'status',
        'payment_method',
        'payment_status',
        'note',
        'delivery_time',
        'estimated_delivery_time'
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'total_amount' => 'decimal:2',
        'delivery_time' => 'datetime',
        'estimated_delivery_time' => 'datetime',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    /**
     * Get the user that owns the order.
     */
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    /**
     * Get the items for the order.
     */
    public function items(): HasMany
    {
        return $this->hasMany(TakeawayOrderItem::class, 'order_id');
    }

    /**
     * Get the tracking events for the order.
     */
    public function tracking(): HasMany
    {
        return $this->hasMany(TakeawayOrderTracking::class, 'order_id');
    }

    /**
     * Get all pending orders.
     */
    public function scopePending($query)
    {
        return $query->where('status', 'Pending');
    }

    /**
     * Get all confirmed orders.
     */
    public function scopeConfirmed($query)
    {
        return $query->where('status', 'Confirmed');
    }

    /**
     * Get all in preparation orders.
     */
    public function scopeInPreparation($query)
    {
        return $query->where('status', 'In Preparation');
    }

    /**
     * Get all out for delivery orders.
     */
    public function scopeOutForDelivery($query)
    {
        return $query->where('status', 'Out for Delivery');
    }

    /**
     * Get all delivered orders.
     */
    public function scopeDelivered($query)
    {
        return $query->where('status', 'Delivered');
    }

    /**
     * Get all cancelled orders.
     */
    public function scopeCancelled($query)
    {
        return $query->where('status', 'Cancelled');
    }

    /**
     * Get all current orders (not delivered or cancelled).
     */
    public function scopeCurrent($query)
    {
        return $query->whereNotIn('status', ['Delivered', 'Cancelled']);
    }

    /**
     * Get all completed orders (delivered or cancelled).
     */
    public function scopeCompleted($query)
    {
        return $query->whereIn('status', ['Delivered', 'Cancelled']);
    }
} 